src/server/common/helpers/auth/session-cookie.js

```diff
@@ -21,13 +21,17 @@ const sessionCookie = {
         },
         keepAlive: true,
         requestDecoratorName: 'sessionCookie',
-        validate: async (request) => {
-          const userSession = await request.getUserSession()
-
+        validate: async (request, session) => {
+          const userSession = await request.getUserSession(session.sessionId)
           if (userSession?.isAuthenticated) {
+            const { scopes, scopeFlags } = await fetchScopes(userSession.token)
             return {
               isValid: true,
-              credentials: await updateUserScope(request, userSession)
+              credentials: {
+                ...userSession,
+                ...(scopeFlags ?? []),
```

Comment:
Pretty sure `scopeFlags` is an object? unless the response from the API has changed? `scopeFlags: { isAdmin: true, isTenant: false }`